src/app/productConfig/modalCtrl.js

```diff
@@ -0,0 +1,40 @@
+import indexOf from 'lodash/indexOf';
+import range from 'lodash/range';
+
+class ModalInstanceCtrl{
+
+  /* @ngInject */
+  constructor($uibModalInstance, designationsService, cartService, productData, itemConfig) {
+    this.$uibModalInstance = $uibModalInstance;
+    this.designationsService = designationsService;
+    this.cartService = cartService;
+    this.productData = productData;
+
+    this.itemConfig = itemConfig;
+  }
+
+  frequencyOrder(f){
+    let order = ['NA', 'MON', 'QUARTERLY', 'ANNUAL', 'SEMIANNUAL'];
+    return indexOf(order, f.name);
+  }
+
+  changeFrequency(productId){
+    this.designationsService.productLookup(productId, true).subscribe((data) => {
+      this.productData = data;
+    });
+  }
+
+  addToCart(){
+    this.cartService.addItem(this.productData.id, this.itemConfig)
+      .subscribe(() => {
+        this.$uibModalInstance.close();
+      });
+  }
+
+  daysInMonth(month){
+    var daysInMonth = new Date(2001, month, 0).getDate();
+    return range(1, daysInMonth + 1);
+  }
+}
+
+module.exports =  ModalInstanceCtrl;
```

Comment:
Does using `export default ModalInstanceCtrl;` work? Also in productConfig.component.js can you import it at the top with `import modalController from './modalCtrl';` and use `modalController` in the `$uibModal` config?

I might also suggest renaming `modalCtrl.js` to `productConfig.modal.js` or something. I think it would make it easier at first glance to know what it is. Let me know if you think of any better naming conventions.